src/object/srv_obj.c

```diff
@@ -1671,6 +1673,58 @@ ds_obj_query_key_handler(crt_rpc_t *rpc)
 		D_ERROR("send reply failed: %d\n", rc);
 }
 
+void
+ds_obj_sync_handler(crt_rpc_t *rpc)
+{
+	struct obj_sync_in	*osi;
+	struct obj_sync_out	*oso;
+	struct ds_cont_hdl	*cont_hdl = NULL;
+	struct ds_cont_child	*cont = NULL;
+	daos_epoch_t		 epoch = daos_ts2epoch();
+	uint32_t		 map_ver = 0;
+	int			 rc;
+
+	osi = crt_req_get(rpc);
+	D_ASSERT(osi != NULL);
+
+	oso = crt_reply_get(rpc);
+	D_ASSERT(oso != NULL);
+
+	if (osi->osi_epoch == 0)
+		oso->oso_epoch = epoch;
+	else
+		oso->oso_epoch = min(epoch, osi->osi_epoch);
```

Comment:
not sure why need the min()? it possibly cause different RDGs' leader do the sync on different epoch.
cannot use the client-side passed in osi_epoch? (may need to set osi_epoch at client if it is zero).

Reply:
Because each RDG has each own independent current highest epoch, then even if client specify the a non-max epoch, we still need to check whether such @epoch is higher than current known highest epoch. It is normal that each RDG sync on each own highest epoch.